tests/Unit/TestHelpers.hpp

```diff
@@ -298,3 +299,21 @@ class DoesThrow {
   DoesThrow& operator=(DoesThrow&&) noexcept(false);
   ~DoesThrow() = default;
 };
+
+template <typename Exception, typename ThrowingFunctor>
+void test_throw_exception(const ThrowingFunctor& func,
+                          const std::string& exception_what_starts_with) {
```

Comment:
Why not take the exact exception type and check the types and message are equal? Then you don't need an explicit template parameter.

Reply:
I'm not sure that I understand you, so let me check. Are you suggesting this
```
template <typename Exception, typename ThrowingFunctor>
void test_throw_exception(const ThrowingFunctor& func,
                          const Exception& expected)
{
  try {
    // throw
  } catch (Exception& e) {
    CHECK(e.what() == expected.what());
  } catch (...) {
    // error
  }
}
```
?

Reply:
Nailed it! Maybe there's a reason not to do that, but that's what I had in mind :)